src/js/server.js

```diff
@@ -110,85 +125,126 @@ class Connection {
+      this.addToMoveListAndBroadcast(message, message.gameID)
     }
 
-    // Consider starting a new game
-    if (Connection.waiting.size == 2 || message.hasComputerOpponent) {
-      let players = Array.from(Connection.waiting.keys())
-      if (message.hasComputerOpponent) {
-        players.push('cpu')
-
-        let cpuPlayer = {
-          'ws': null, //unneeded maybe I should delete
-          'hasComputerOpponent': false,  //unneeded maybe I should delete
-          'address': null,  //unneeded maybe I should delete
-          'name': 'cpu',
-          'deck': DECKS[0], //the bibot deck
-        }
-        Connection.all.set('cpuAddress', cpuPlayer)
-      }
-      let gameID = Math.floor(Math.random() * 1000000)
-      console.log(`starting ${players[0]} vs ${players[1]}. gameID: ${gameID}`)
-      let start = { op: "start", players, gameID }
-      this.broadcast(start)
-      Connection.waiting.clear()
-
-      // everyone starts with three cards
-      for (let i = 0; i < STARTING_HAND_SIZE; i++) {        
-        this.everyoneDraws()
-      }
-      
-      // you are always drawing cards in spacetime
-      this.drawLoop = setInterval(() => {
-        this.tickTurn();
-      }, DRAW_MS);
-
-      this.currentGameSecond = DRAW_MS / 1000;
-      this.timeLoop = setInterval(() => {
-        this.currentGameSecond--;
-        this.broadcast({ op: "tickTime", time: this.currentGameSecond, player: "no_player"})
-      }, 1000);
+  }
+
+  endGame(gameID) {
+    clearInterval(Connection.timeLoops.get(gameID))
+    clearInterval(Connection.drawLoops.get(gameID))
+    // might not need this
+    Connection.currentGameSeconds.set(gameID, null)
+  }
+
+  // Deal cards, and start game loop.
+  startGame(message, gameID) {
+     
+    // set a blank list for the moves for the game
+    Connection.games.set(gameID, [])
+
+    // add a computer player if needed  
+    let players = Array.from(Connection.waiting.keys())
+    if (message.hasComputerOpponent) {
+      let cpuName = 'cpu' + gameID
+      players.push(cpuName)
+      let cpuPlayer = this.defaultComputerPlayer(gameID)
+      Connection.all.set('cpuAddress', cpuPlayer)
+    }
+   
+    // send the start move
+    console.log(`starting ${players[0]} vs ${players[1]}. gameID: ${gameID}`)
+    let start = { op: "start", players, gameID}
+    this.addToMoveListAndBroadcast(start, gameID)
+    Connection.waiting.clear()
+    
+    // everyone starts with three cards
+    for (let i = 0; i < STARTING_HAND_SIZE; i++) {        
+      this.everyoneDraws(gameID)
+    }
+    
+    // Players draw and receive mana occasionally.
+    // Some cards played trigger on timers.
+    this.startGameLoop(gameID)
 
+  }
+
+  // A computer player that pumps out simple attacking cards.
+  defaultComputerPlayer(gameID) {
+    return {
+      'name': 'cpu' + gameID,
+      'gameID': gameID,
+      'deck': DECKS[0], //the bibot deck
+      'ws': null, //unneeded maybe I should delete
```

Comment:
it's not just unneeded, it's probably an error if this gets accessed. I would suggest either deleting it or making it an obviously-filler value like "No Websocket"